source/backend/tests/unit/api/Services/ProjectServiceTest.cs

```diff
@@ -196,5 +196,42 @@ public async void Add_Project_Success()
             result.Should().NotBeNull();
             repository.Verify(x => x.Add(It.IsAny<PimsProject>()), Times.Once);
         }
+
+        [Fact]
+        public void Get_ProjectById_ShouldFail_NotAuthorized()
+        {
+            // Arrange
+            var helper = new TestHelper();
+            var user = PrincipalHelper.CreateForPermission();
+            var service = helper.Create<ProjectService>(user);
+
+            var repository = helper.GetService<Mock<IProjectRepository>>();
+
+            // Act
+            Func<Task> act = async () => await service.GetById(It.IsAny<long>());
+
+            // Assert
+            act.Should().Throw<NotAuthorizedException>();
+            repository.Verify(x => x.Get(It.IsAny<long>()), Times.Never);
```

Comment:
We are no longer asserting that repos are never called when exception was thrown by service, please remove this line